packages/core/src/external/carequality/dq/process-incoming-dq.ts

```diff
@@ -54,44 +27,16 @@ export async function processIncomingRequest(
     };
 
     return response;
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  } catch (error: any) {
-    switch (error.constructor) {
-      case XDSUnknownPatientId:
-        return constructErrorResponse(
-          payload,
-          DQ_CODE_SYSTEM_REQUIRED_ERROR,
-          "XDSUnknownPatientId",
-          error.message
-        );
-      case XDSUnknownCommunity:
-        return constructErrorResponse(
-          payload,
-          DQ_CODE_SYSTEM_REQUIRED_ERROR,
-          "XDSUnknownCommunity",
-          error.message
-        );
-      case XDSMissingHomeCommunityId:
-        return constructErrorResponse(
-          payload,
-          DQ_CODE_SYSTEM_REQUIRED_ERROR,
-          "XDSMissingHomeCommunityId",
-          error.message
-        );
-      case XDSRegistryError:
-        return constructErrorResponse(
-          payload,
-          DQ_CODE_SYSTEM_REQUIRED_ERROR,
-          "XDSRegistryError",
-          error.message
-        );
-      default:
-        return constructErrorResponse(
-          payload,
-          DQ_CODE_SYSTEM_REQUIRED_ERROR,
-          "Internal Server Error",
-          "Unknown Error: Contact Metriport Support for assistance"
-        );
+  } catch (error) {
+    if (
+      error instanceof XDSUnknownPatientId ||
+      error instanceof XDSUnknownCommunity ||
+      error instanceof XDSMissingHomeCommunityId ||
+      error instanceof XDSRegistryError
```

Comment:
IIRC there's a super class that we could use here instead of checking each subclass, no? deja-vu...